ctd/__init__.py

```diff
@@ -67,19 +81,21 @@
 # Attach methods.
 Index.asof = asof
 
-Series.plot = plot
 Series.split = split
 Series.smooth = smooth
 Series.despike = despike
 Series.bindata = bindata
 Series.press_check = press_check
+if HAS_MATPLOTLIB:
+    Series.plot = plot
 
 DataFrame.split = split
 DataFrame.from_cnv = staticmethod(from_cnv)
 DataFrame.from_btl = staticmethod(from_btl)
 DataFrame.from_edf = staticmethod(from_edf)
 DataFrame.from_fsi = staticmethod(from_fsi)
-DataFrame.plot_vars = plot_vars
 DataFrame.press_check = press_check
-DataFrame.get_maxdepth = get_maxdepth
-DataFrame.plot_section = plot_section
+if HAS_MATPLOTLIB:
```

Comment:
What is your motivation for this change?

In my original design matplotlib was as mandatory dependency. I'm opened to the idea of making it option, just not sure if the extra lines of code are worth it since mpl is pretty easy to install and widely available these days.

Reply:
There is an annoying issue when running matplotlib under python as a framework build on MacOS. Anaconda is built this way and I get an import error when trying to use ctd. The extra code is there to avoid the error and provide a warning instead. There is more info about the issue here: https://matplotlib.org/faq/osx_framework.html

This is the error I get on with setup (Anaconda, python3, MacOS):

In [1]: import ctd
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-1-6e356f12f043> in <module>()
----> 1 import ctd

~/anaconda3/lib/python3.6/site-packages/ctd/__init__.py in <module>()
     11     rosette_summary
     12 )
---> 13 from .plotting import (
     14     extrap_sec,
     15     gen_topomask,

~/anaconda3/lib/python3.6/site-packages/ctd/plotting.py in <module>()
      3 from __future__ import absolute_import, division, print_function
      4 
----> 5 import matplotlib.pyplot as plt
      6 
      7 import mpl_toolkits.axisartist as AA

~/anaconda3/lib/python3.6/site-packages/matplotlib/pyplot.py in <module>()
    113 ## Global ##
    114 
--> 115 _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
    116 
    117 _IP_REGISTERED = None

~/anaconda3/lib/python3.6/site-packages/matplotlib/backends/__init__.py in pylab_setup(name)
     60     # imports. 0 means only perform absolute imports.
     61     backend_mod = __import__(backend_name, globals(), locals(),
---> 62                              [backend_name], 0)
     63 
     64     # Things we pull in from all backends

~/anaconda3/lib/python3.6/site-packages/matplotlib/backends/backend_macosx.py in <module>()
     15 
     16 import matplotlib
---> 17 from matplotlib.backends import _macosx
     18 
     19 from .backend_agg import FigureCanvasAgg

RuntimeError: Python is not installed as a framework. The Mac OS X backend will not be able to function correctly if Python is not installed as a framework. See the Python documentation for more information on installing Python as a framework on Mac OS X. Please either reinstall Python as a framework, or try one of the other backends. If you are using (Ana)Conda please install python.app and replace the use of 'python' with 'pythonw'. See 'Working with Matplotlib on OSX' in the Matplotlib FAQ for more information.

I normally use Anaconda with pythonw but there is a specific project where that's not possible.